geomstats/geometry/siegel.py

```diff
@@ -0,0 +1,614 @@
+        ----------
+        tangent_vec : array-like, shape=[..., n, n]
+            Tangent vector at zero.
+        base_point : array-like, shape=[..., n, n]
+            Point on the Siegel space.
+
+        Returns
+        -------
+        tangent_vec_at_base_point : array-like, shape=[..., n, n]
+            Tangent vector at the base point.
+        """
+        data_type = (tangent_vec + base_point).dtype
+        identity = create_identity_mat(tangent_vec.shape, dtype=data_type)
+        base_point_transconj = ComplexMatrices.transconjugate(base_point)
+        aux_1 = gs.einsum("...ij,...jk->...ik", base_point, base_point_transconj)
+        aux_2 = gs.einsum("...ij,...jk->...ik", base_point_transconj, base_point)
+        aux_3 = identity - aux_1
+        aux_4 = identity - aux_2
+        factor_1 = HermitianMatrices.powerm(aux_3, 1 / 2)
+        factor_3 = HermitianMatrices.powerm(aux_4, 1 / 2)
+        prod_1 = gs.einsum("...ij,...jk->...ik", factor_1, tangent_vec)
+        tangent_vec_at_base_point = gs.einsum("...ij,...jk->...ik", prod_1, factor_3)
+        return tangent_vec_at_base_point
+
+    def log(self, point, base_point):
+        """Compute the Siegel logarithm map.
+
+        Compute the Riemannian logarithm at point base_point
+        of point wrt the metric defined in inner_product.
+        Return a tangent vector at point base_point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Point on the Siegel space.
+        base_point : array-like, shape=[..., n, n]
+            Point on the Siegel space.
+
+        Returns
+        -------
+        log : array-like, shape=[..., n, n]
+            Riemannian logarithm at the base point.
+        """
+        point_at_zero = self.isometry(point=point, point_to_zero=base_point)
+
+        logarithm_at_zero = self.log_at_zero(point_at_zero)
+
+        log = self.tangent_vec_from_zero_to_base_point(
+            tangent_vec=logarithm_at_zero, base_point=base_point
+        )
+
+        return log
+
+    def squared_dist(self, point_a, point_b):
+        """Compute the Siegel squared distance.
+
+        Compute the Riemannian squared distance between point_a and point_b.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., n, n]
+            Point.
+        point_b : array-like, shape=[..., n, n]
+            Point.
+
+        Returns
+        -------
+        squared_dist : array-like, shape=[...]
+            Riemannian squared distance.
+
+        References
+        ----------
+        The Kahler mean of Block-Toeplitz matrices
+        with Toeplitz structured blocks
+        B. Jeuris and R. Vandebril
+        2016
+        https://epubs.siam.org/doi/pdf/10.1137/15M102112X
+        3.2. A second generalized transformation.
+        """
+        if len(point_a.shape) > len(point_b.shape):
+            point_a, point_b = point_b, point_a
+
+        data_type = (point_a + point_b).dtype
```

Comment:
use `point_a`. no need to create a variable.

Reply:
Done.